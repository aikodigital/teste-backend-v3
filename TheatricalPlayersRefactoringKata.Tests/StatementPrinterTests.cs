using System;
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using Xunit;

namespace TheatricalPlayersRefactoringKata.Tests;

public class StatementPrinterTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TestStatementExampleLegacy()
    {
        var plays = new Dictionary<string, Play>();
        plays.Add("hamlet", new Play("Hamlet", 4024, "tragedy"));
        plays.Add("as-like", new Play("As You Like It", 2670, "comedy"));
        plays.Add("othello", new Play("Othello", 3560, "tragedy"));

        Invoice invoice = new Invoice(
            "BigCo",
            new List<Performance>
            {
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40),
            }
        );

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.Print(invoice, plays);

        Approvals.Verify(result);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TestTextStatementExample()
    {
        var plays = new Dictionary<string, Play>();
        plays.Add("hamlet", new Play("Hamlet", 4024, "tragedy"));
        plays.Add("as-like", new Play("As You Like It", 2670, "comedy"));
        plays.Add("othello", new Play("Othello", 3560, "tragedy"));
        plays.Add("henry-v", new Play("Henry V", 3227, "history"));
        plays.Add("john", new Play("King John", 2648, "history"));
        plays.Add("richard-iii", new Play("Richard III", 3718, "history"));

        Invoice invoice = new Invoice(
            "BigCo",
            new List<Performance>
            {
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40),
                new Performance("henry-v", 20),
                new Performance("john", 39),
                new Performance("henry-v", 20)
            }
        );

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.Print(invoice, plays);

        Approvals.Verify(result);
    }
}
