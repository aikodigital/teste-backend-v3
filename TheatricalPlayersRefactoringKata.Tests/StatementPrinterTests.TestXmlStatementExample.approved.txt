<?xml version="1.0" encoding="utf-8"?>
<Statement xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Customer>BigCo</Customer>
  <Items>
    <Item>
      <AmountOwed>650</AmountOwed>
      <EarnedCredits>25</EarnedCredits>
      <Seats>55</Seats>
    </Item>
    <Item>
      <AmountOwed>547</AmountOwed>
      <EarnedCredits>12</EarnedCredits>
      <Seats>35</Seats>
    </Item>
    <Item>
      <AmountOwed>456</AmountOwed>
      <EarnedCredits>10</EarnedCredits>
      <Seats>40</Seats>
    </Item>
    <Item>
      <AmountOwed>705.4</AmountOwed>
      <EarnedCredits>0</EarnedCredits>
      <Seats>20</Seats>
    </Item>
    <Item>
      <AmountOwed>931.6</AmountOwed>
      <EarnedCredits>9</EarnedCredits>
      <Seats>39</Seats>
    </Item>
    <Item>
      <AmountOwed>705.4</AmountOwed>
      <EarnedCredits>0</EarnedCredits>
      <Seats>20</Seats>
    </Item>
  </Items>
  <AmountOwed>3995.4</AmountOwed>
  <EarnedCredits>56</EarnedCredits>
</Statement>